.circleci/scripts/git-diff-develop.ts

```diff
@@ -0,0 +1,94 @@
+import { promisify } from 'util';
+import { exec as execCallback } from 'child_process';
+import fs from 'fs';
+import path from 'path';
+
+const exec = promisify(execCallback);
+
+/**
+ * Fetches the git repository with a specified depth.
+ *
+ * @param {number} depth - The depth to use for the fetch command.
```

Comment:
Nit: We should omit the types from these comments in TypeScript files, since the types are already documented in the type signature. They can make the comments overly verbose, and can easily get out of sync.

e.g.
```suggestion
 * @param depth - The depth to use for the fetch command.
```

Technically for TypeScript files we're following something closer to the TSDoc standard: https://tsdoc.org
It's slightly different from JSDoc, it has fewer tags, and some extra ones specific to TypeScript.

Reply:
ooh thank you for the detailed explanation Mark :heart: make sense to me